pkg/datalayer/datalayer.go

```diff
@@ -21,10 +22,50 @@ func VolumeExists(basePath string, volumeName string) bool {
 	return err == nil
 }
 
+func ActiveVolume(basePath string) (string, error) {
+	currentVolumeJsonPath := filepath.FromSlash(basePath + "/" + "current_volume.json")
+	file, err := os.Open(currentVolumeJsonPath)
+	if err != nil {
+		return "", err
+	}
+	defer file.Close()
+	decoder := json.NewDecoder(file)
+	var store map[string]interface{}
+	err = decoder.Decode(&store)
+	if err != nil {
+		return "", err
+	}
+	return store["current_volume"].(string), nil
+}
+
+func setActiveVolume(basePath, volumeName string) error {
+	currentVolumeJsonPath := filepath.FromSlash(basePath + "/" + "current_volume.json")
```

Comment:
reasoning about why basePath will already have been created here: MkdirAll must have run, because you can't switch to a volume before creating it, and creating a volume implicitly creates basePath because it's a subdirectory of it and MkdirAll is like `mkdir -p`. not actionable, just reasoning about why the code is correct.